keps/sig-api-machinery/3352-aggregated-discovery/kep.yaml

```diff
@@ -24,16 +25,16 @@ prr-approvers:
   - "@deads2k"
 
 # The target maturity stage in the current dev cycle for this KEP.
-stage: alpha
+stage: beta
```

Comment:
@smarterclayton my understanding was that this still started with an alpha featuregate, but a beta serialization.  So effectively it was off-by-default, but stable for clients if available.

Reply:
I don't really understand an alpha feature gate gating a beta API... off by default seems reasonable, but we can do a beta off-by-default gate

Reply:
I think from a clarity perspective "off by default" is the first phase of non-resource APIs and the next phase is "on by default and can't evolve it" (i.e. metrics is 'stable' once we turn it on by default since we have no versioning scheme).  We have no question here whether we want to evolve our discovery API to add "at once" retrieval (so it's not alpha because we need feedback).   We have a versioning scheme here, so "off by default" but "beta" doesn't prevent us from evolving or killing.  

I think based on what we've demonstrated we would be comfortable with users turning this flag on within one release, because the mechanisms give us the ability to evolve, and if we decided not to graduate it would be no different from any other beta feature being dropped.  So beta gate is based on assessed confidence in the feature.

Reply:
> I don't really understand an alpha feature gate gating a beta API.

The featuregate is also gating the new controllers that are providing the data and the (hopefully settling) etag calculation across multiple HA servers.  The serialization is based on experience, so beta stability serialization is logical, but the code providing the data to serialize are fully novel.

If we're going to require an "on by default beta" before going GA, what benefit have we gained by exposing unvetted controllers as beta in their first release?  Calling our off-by-default alpha is consistent, doesn't preclude using a more stable serialization, and keeps maturity graduation consistent.  Vendors can (and do) turn alpha features on for testing and maybe even production: see p&f and graceful node shutdown.


> So beta gate is based on assessed confidence in the feature.

Is this confidence a thing that any sig can say about any feature they wish?  Choosing a couple at semi-random
1. pleg performance improvements
2. node swap
3. kubelet credential providers

Reply:
> If we're going to require an "on by default beta" before going GA, what benefit have we gained by exposing unvetted controllers as beta in their first release?

I think the plan was to do "off-by-default" beta followed by "on-by-default" GA.

> Is this confidence a thing that any sig can say about any feature they wish?

I think this is a combination of the 9 points enumerated above.